src/libMessage/Messenger.cpp

```diff
@@ -658,12 +658,22 @@ namespace
             return false;
         }
 
-        if ((announcement.consensusinfo().blockhash().size()
-             != blockHash.size())
+        std::vector<unsigned char> remoteBlockHash;
```

Comment:
Hmm I think going through `remoteBlockHash` is unnecessary. We can just say `if ((announcement.consensusinfo().blockhash().size() != blockHash.size()) || !equal(blockHash.begin(), blockHash.end(), announcement.consensusinfo().blockhash().begin()))`

Reply:
In the event that the check fails and you need to log the hash, then yeah I think we can do `remoteBlockHash` inside the LOG_GENERAL, but not until the check fails

Reply:
I find if we don't covert it to std::vector<unsigned char> remoteBlockhash first, this comparison always fail, i don't know why.

Reply:
Looks like this will make it work: `if (equal(blockHash.begin(), blockHash.end(), announcement.consensusinfo().blockhash().begin(), [](const unsigned char left, const char right) -> bool { return left == (unsigned char)right; }))`

Reply:
Let me try this solution.

Reply:
Latest commit looks ok. As discussed, we will do another improvement later on (#648)